### PR TITLE
code cleanup, add unit tests, hook up eslint and prettier

### PR DESCRIPTION
**Pull request by jetersen**
reduces branching and cleans up the nested if statements and for loop.

Aligns the various logging between `linux`, `windows` and `darwin` brances.

hook prettier into eslint.

align trailing comma style in prettier to existing `updatecliPackages` array in index.js

~removed `core.addPath` in `updatecliDownload` as it is called in `updatecliVersion`.~ Could potentially also remove the `tool.find` and `core.addPath` in `updatecliVersion` and keep `core.addPath` in `updatecliDownload` - Done

use top level await which node16 does support.

### CODE REVIEW

**Comment by jetersen** [comment]
actually to my knowledge github actions are not offered in `arm64` arch. So we could cleanup the logic even more.

File: src/main.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+export async function updatecliExtract(downloadPath, downloadUrl) {
+  if (downloadUrl.endsWith('.tar.gz')) {
+    return tool.extractTar(downloadPath)
+  } else if (downloadUrl.endsWith('.zip')) {
+    return tool.extractZip(downloadPath)
+  } else {
+    throw new Error(`Unsupported archive type: ${downloadUrl}`)
+  }
+}
```
Comment (jetersen) [inline src/main.js:14]: FYI previous logic for `darwin` did not make sense as Xar is not supported on Mac 🤣

https://github.com/actions/toolkit/blob/c5278cdd088a8ed6a87dbd5c80d7c1ae03beb6e5/packages/tool-cache/src/tool-cache.ts#L297

File: package.json [hunk 2]
```diff
@@ -6,7 +6,7 @@
   "scripts": {
     "lint": "eslint .",
     "prepare": "ncc build index.js -o dist --source-map --license licenses.txt",
-    "test": "jest",
+    "test": "cross-env NODE_OPTIONS=--experimental-vm-modules jest",
```
Comment (jetersen) [inline package.json:9]: currently needed for jest when using ESM instead of commonjs modules

File: tests/main.test.js [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+const directory = path.dirname(url.fileURLToPath(import.meta.url))
+
+const cachePath = path.join(directory, 'CACHE')
+const temporaryPath = path.join(directory, 'TEMP')
+// Set temp and tool directories before importing (used to set global state)
+process.env['RUNNER_TEMP'] = temporaryPath
+process.env['RUNNER_TOOL_CACHE'] = cachePath
+
+const version = 'v0.28.0'
```
Comment (jetersen) [inline tests/main.test.js:19]: Could potentially read the version from the `action.yml`

File: tests/main.test.js [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+const fakePlatformArch = (fakePlatform, fakeArch) => {
+  Object.defineProperty(process, 'platform', {
+    value: fakePlatform,
+  })
+  Object.defineProperty(process, 'arch', {
+    value: fakeArch,
+  })
+}
+
+describe('main', () => {
```
Comment (jetersen) [inline tests/main.test.js:43]: 100% code coverage 👏

**Comment by jetersen** [comment]
prefer squash :)

Potentially pr title could be better 🤷‍♂️

File: tests/main.test.js (hunk 3 above)
Comment (lemeurherve) [reply tests/main.test.js:20]: Or added as target in this manifest: https://github.com/updatecli/updatecli-action/blob/v2/updatecli/updatecli.d/updatecli.yaml#L36-L43

**Review by lemeurherve — APPROVED** [review]
LGTM, thanks!

File: tests/main.test.js (hunk 3 above)
Comment (jetersen) [reply tests/main.test.js:20]: added target in aae16d3

**Review by lemeurherve — APPROVED** [review]

File: src/main.js (hunk 1 above)
Comment (olblak) [reply src/main.js:14]: Thanks for the information.
As a none mac user, I just followed some documentation :).

**Comment by olblak** [comment]
Thank you very much @jetersen for this pullrequest. Your Javascript skills are from very far better than mine :) and I must admit I gave up before getting to this level.

**Comment by jetersen** [comment]
Release drafter has me sharpening my JavaScript skills 😅

**Comment by jetersen** [comment]
Also! Very important GitHub copilot was very good at suggesting once it had context. I don't know what I would do without it. 😅

**Comment by jetersen** [comment]
@lemeurherve @olblak 

This updatecli replaceWith looks like it foobars:
https://github.com/updatecli/updatecli-action/runs/7479718688?check_suite_focus=true#step:5:87

File: updatecli/updatecli.d/updatecli.yaml [hunk 5]
```diff
@@ -41,4 +41,11 @@ targets:
             key: inputs.version.default
         scmid: default
         sourceid: updatecli
+    tests:
+        name: Bump updatecli version in main.test.js
+        kind: file
+        spec:
+            file: tests/main.test.js
+            matchPattern: const version = '(.*)'
+            replaceWith: const version = '{{ source "updatecli" }}'
```
Comment (jetersen) [inline updatecli/updatecli.d/updatecli.yaml:50]: should have been `replacepattern`

**Pull request merged**